src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs

```diff
@@ -137,6 +138,85 @@ public static IEnumerable<OdcmMethod> GetMethods(this OdcmModel model)
             return model.GetEntityTypes().SelectMany(entityType => entityType.Methods);
         }
 
+        /// <summary>
+        /// This extension method determines whether the current type needs to be disambiguated.
+        /// If the current type needs dismabiguation, we then provide a fully qualified 
+        /// import statement for the model.
+        /// This currently operates only on entities whose name ends in "Request". We do this
+        /// because every entity results in a request object. For example, assume we have the following
+        /// two entities: timeOff and timeOffRequest. The timeOff entity will trigger the generation
+        /// of classes named model.timeOff and request.timeOffRequest. The timeOffRequest entity
+        /// will trigger the generation of a model.timeOffRequest and request.timeOffRequestRequest.
+        /// The request.timeOffRequest and model.timeOffRequest classes will result in a name collision in
+        /// a few files.
+        /// Assumptions: 1) host.CurrentType is an OdcmProperty.
+        /// </summary>
+        /// <param name="host">The T4Host that orchestrates applying templates to the OdcmModel.</param>
+        /// <returns>A boolean value that indicates whether the current type needs to be disambiguated.</returns>
+        public static bool DoesCurrentTypeNeedDisambiguation(this CustomT4Host host)
+        {
+            // At this point this is only applicable to OdcmProperty.
+            // Challenging this assumption will require a lot more investigation.
+            if (!(host.CurrentType is OdcmProperty))
+                return false; 
+
+            // We only support "Request" dismabiguation at this point. Check whether the
+            // current type ends in "Request".
+            var requestSuffix = "Request";
+            var currentTypeName = (host.CurrentType as OdcmProperty).Type.Name;
+            int index = currentTypeName.IndexOf(requestSuffix);
```

Comment:
This logic may get false positives when "Request" is followed by something else, e.g. "somethingRequestSomething" (though it may be very unlikely). I think we should use `EndsWith()` instead to futureproof.

Or:
```
var shouldDisambiguate = classes.Where(entity => entity.Kind == OdcmClassKind.Entity
                                                       && entity.Name + "Request" == currentTypeName).Any();
```

But I prefer `EndsWith()` as it entails less string operations.